iceoryx_utils/cmake/IceoryxPlatform.cmake

```diff
@@ -112,7 +116,8 @@ if(sanitize)
 
     if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU" OR CMAKE_CXX_COMPILER_ID STREQUAL "Clang" OR CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang")
         set(ICEORYX_SANITIZER_COMMON_FLAGS -fno-omit-frame-pointer -fno-optimize-sibling-calls -g -O1)
-
+        set(OPTIMIZE_OPTIONS -g -O1)
```

Comment:
Even better would be to disable all optimization in sanitize mode to really catch all errors and add all possible debug symbols. I would suggest `set(OPTIMIZE_OPTIONS -ggdb -O0)`

Reply:
i would prefer to have it `-g` only for compatibility to other compilers